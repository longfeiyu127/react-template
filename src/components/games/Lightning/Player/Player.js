import React from 'react';
// import Tictactoe from './tictactoe'
// import Fivechess from './fivechess'
import './Player.less';
import PlayerConfig from '../../../../service/config/Lightning/PlayerConfig';

const { size } = PlayerConfig;

class Player extends React.Component {
  constructor() {
    super();
    this.state = {
      // playertop: '',
      // left: '',
    };
  }

  render() {
    // eslint-disable-next-line
    const { position } = this.props;
    // console.log(position);
    return (
      <div
        className="g-lig-player"
        style={{
          top: `${position.top}rem`,
          left: `${position.left}rem`,
          height: `${size.height}rem`,
          width: `${size.width}rem`
        }}
      >
        {/* style={{fontWeight: (move === stepNumber) ? 900 : 500, cursor: 'pointer'}} */}
        <main className="player-main" />
      </div>
    );
  }
}

export default Player;
