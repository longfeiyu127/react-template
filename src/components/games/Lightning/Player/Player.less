.g-lig-player{
  height: 0.5rem;
  width: 0.5rem;
  position: absolute;
  .player-main{
    height: 100%;
    background: url('../../../../assets/images/games/lightning/player/plane01.png') no-repeat center/100% 100%;
  }
}
