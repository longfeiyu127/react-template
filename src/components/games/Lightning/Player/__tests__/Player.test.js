import React from 'react';
import renderer from 'react-test-renderer';
// import Enzyme, { mount } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import { Player } from '../Player';

// Enzyme.configure({ adapter: new Adapter() });

describe('<Player />', () => {
  it('component render', () => {
    const position = {
      top: 0.1,
      left: 0.1
    };
    let player = renderer.create(<Player position={position} />).toJSON();
    expect(player).toMatchSnapshot();

    player = renderer.create(<Player position={position} />).toJSON();
    expect(player).toMatchSnapshot();
  });

  // it('componentDidMount', () => {
  //   const fn = jest.fn();
  //   const position = {
  //     top: 0.1,
  //     left: 0.1
  //   };
  //   const player = mount(<Player position={position} ControllPlayer={fn} />);
  //   player.instance().componentDidMount();
  //   expect(fn.mock.calls.length).toBe(1);
  // });
});
