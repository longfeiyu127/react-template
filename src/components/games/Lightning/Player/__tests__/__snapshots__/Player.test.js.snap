// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> component render 1`] = `
<div
  className="g-lig-player"
  style={
    Object {
      "height": "0.56rem",
      "left": "0.1rem",
      "top": "0.1rem",
      "width": "0.8rem",
    }
  }
>
  <main
    className="player-main"
  />
</div>
`;

exports[`<Player /> component render 2`] = `
<div
  className="g-lig-player"
  style={
    Object {
      "height": "0.56rem",
      "left": "0.1rem",
      "top": "0.1rem",
      "width": "0.8rem",
    }
  }
>
  <main
    className="player-main"
  />
</div>
`;
